### DISCUSSION TITLE
Fancybox for Waymark Maps

### DISCUSSION THREAD

**Discussion by selcukdere** [OpenGIS/Waymark#38] [Q&A]
Could anyone tell me how to activate fancybox in Waymark map. Thanks

**Comment by morehawes** [comment]
Hi @selcukdere,

Thanks for reaching out :) Improving the handling of images and the Image Gallery are both on my list, as I know there is room for improvement here!

Could you provide some more details about what you are hoping to achieve?...

- Are you looking for Fancybox gallery everywhere a Map is embedded (i.e. using the Shortcode) or just on the Map Details page?
- Would the Fancybox be triggered by clicking on the existing Image Gallery thumbail (at the bottom of the Map), or only when clicking the larger version in the Info popup (currently links to full-size image)?
- Are you looking for just the images to be displayed in the Lightbox/Fancybox, or would you like additional text, location info etc?
- When "cycling" though Images - i.e. all Images for one Map would be grouped as a slideshow, are you wanting the Map to adjust it's view with each Image to show (i.e. "fly to") that location on the Map where the Image is located?

Thanks again for your input, this has been on my to-do list for a while!

Joe

**Comment by selcukdere** [comment]
Hi @morehawes, existing image gallery works great for centering the pin on the map and It will be good if direct link of image to be displayed in fancybox window.

And also I cannot attach the **waymark-js.min.js** to the page. 

Wordpress using **waymark-js.min.js?ver=1.1.4** file but I cannot edit this attached file
`<script src="https://www.example.com/wp-content/plugins/waymark/waymark-js/dist/js/**waymark-js.min.js?ver=1.1.4**" id="waymark-js-js"></script>`

When I change the waymark-js.min.js file it doesnt apply to the site. Can you give any solution for this.


By the way adding _data-fancybox="gallery"_ to the waymark-js.min.js file makes the fancybox working in maps

`<a href="${t.properties.image_large_url}" data-fancybox="gallery" style="background-image:url(${s})">`

`attr({href:e.properties.image_large_url,data-fancybox="gallery",style:"background-image:url`

Thanks for your help.

**Comment by morehawes** [comment]
Hi @selcukdere,

> existing image gallery works great for centering the pin on the map and It will be good if direct link of image to be displayed in fancybox window.

Thanks for the extra information, so the trigger would be clicking to enlarge the image in the info window, which makes sense to me :)

This has been requested before and as there are lots of different lightbox libraries (WordPress even comes bundled with one) I will need to think about the best way to handle this.

Currently these images are added dynamically (by JavaScript/Leaflet) when a user clicks an Overlay. This makes them a little tricky to "group" into a lightbox - but I think it wouldn't be too much work to structure things so they are friendlier to lightbox libraries.

> When I change the waymark-js.min.js file it doesnt apply to the site. Can you give any solution for this.

If you are looking to modify Waymark you will need to edit the source files and run the build script as described [here](https://github.com/OpenGIS/Waymark#development).

I am currently traveling at the moment, but will try to look in to a solution for this once I get more time.

Cheers,

Joe

**Reply by selcukdere** [reply]
Thanks for your help. Have a good trip.
